src/ai/susi/server/api/aaa/GetApiKeys.java

```diff
@@ -0,0 +1,80 @@
+/**
+ *  GetApiKeys
+ *  Copyright by Praduman @PrP-11 on 29/07/18.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.json.JsonTray;
+import ai.susi.server.*;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import ai.susi.server.APIException;
+import ai.susi.server.APIHandler;
+import ai.susi.server.AbstractAPIHandler;
+import ai.susi.server.Query;
+import ai.susi.server.ServiceResponse;
+import ai.susi.server.UserRole;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This Servlet gives a API Endpoint to add, modify and delete different API keys used by SUSI.
+ * It requires user role to be ANONYMOUS or above ANONYMOUS
+ * example:
+ * http://localhost:4000/aaa/apiKeys.json
+ */
+
+public class GetApiKeys extends AbstractAPIHandler implements APIHandler {
+
+    @Override
+    public String getAPIPath() {
+        return "/aaa/getApiKeys.json";
+    }
+
+    @Override
+    public UserRole getMinimalUserRole() {
+        return UserRole.ANONYMOUS;
```

Comment:
`ANONYMOUS` ? It should strictly be `ADMIN` and above.

Reply:
Refer to this - https://docs.google.com/spreadsheets/d/116QIngPEW2QI_p6YEwcj_vQ0dVMTzWZTB-wjWC4s_2s/edit#gid=1053849824

Reply:
@akshatnitd Please see my comment above

Reply:
@akshatnitd Currently API keys are hardcoded on the clients and therefore we need to implement them on server so that admins can add and edit API keys from admin panel itself. And regarding use-case for `ANONYMOUS` users, they require the CAPTCHA_KEY to signup with susi for example. Also I have implemented this as outlined in the sheet. In the sheet itself `ADMIN` and `SUPERADMIN` is required to edit keys only